tests/cupyx_tests/scipy_tests/linalg_tests/test_decomp_lu.py

```diff
@@ -0,0 +1,63 @@
+import unittest
+
+import numpy
+try:
+    import scipy.linalg
+
+    scipy_available = True
+except ImportError:
+    scipy_available = False
+
+import cupy
+from cupy import cuda
+from cupy import testing
+import cupyx.scipy.linalg
+
+
+@unittest.skipUnless(
+    cuda.cusolver_enabled, 'Only cusolver in CUDA 8.0 is supported')
+@unittest.skipUnless(scipy_available, 'requires scipy')
+@testing.gpu
+@testing.fix_random()
+class TestLUFactor(unittest.TestCase):
+
+    @testing.for_float_dtypes(no_float16=True)
+    def check_x(self, array, dtype):
+        a_cpu = numpy.asarray(array, dtype=dtype)
+        a_gpu = cupy.asarray(array, dtype=dtype)
+        result_cpu = scipy.linalg.lu_factor(a_cpu)
+        result_gpu = cupyx.scipy.linalg.lu_factor(a_gpu)
+        self.assertEqual(result_cpu[0].dtype, result_gpu[0].dtype)
+        self.assertEqual(result_cpu[1].dtype, result_gpu[1].dtype)
+        cupy.testing.assert_allclose(result_cpu[0], result_gpu[0], atol=1e-5)
+        cupy.testing.assert_array_equal(result_cpu[1], result_gpu[1])
+
+    def test_lu_factor(self):
+        self.check_x(numpy.random.randn(1, 1))
+        self.check_x(numpy.random.randn(2, 2))
+        self.check_x(numpy.random.randn(3, 3))
+        self.check_x(numpy.random.randn(5, 5))
+
+
+@unittest.skipUnless(
+    cuda.cusolver_enabled, 'Only cusolver in CUDA 8.0 is supported')
+@unittest.skipUnless(scipy_available, 'requires scipy')
+@testing.gpu
+@testing.parameterize(
+    {'trans': 0},
+    {'trans': 1},
```

Comment:
Why not adding a test for `trans = 2`?
How about parameterizing shapes?
```py
@parameterize(*testing.product({
    'trans': [0, 1, 2],
    'shapes': [((4, 4), (4,)), ((5, 5), (5, 2))],
}))
```

Reply:
I didn't tested for `trans = 2` (`CUBLAS_OP_C`) just because I thought conjugate transposition (`CUBLAS_OP_C`) is meaningless for real matrix. But it seems that `trans = 2` works without any problem (after I fixed my typo `CUBLAS_OP_H`). So I'll add a test case.

Also I'll parameterize shapes as you suggested.

Thanks!

Reply:
done.